<!--: spam
Content-Type: text/html

<body bgcolor="#f0f0f8"><font color="#f0f0f8" size="-5"> -->
<body bgcolor="#f0f0f8"><font color="#f0f0f8" size="-5"> --> -->
</font> </font> </font> </script> </object> </blockquote> </pre>
</table> </table> </table> </table> </table> </font> </font> </font><body bgcolor="#f0f0f8">
<table width="100%" cellspacing=0 cellpadding=2 border=0 summary="heading">
<tr bgcolor="#6622aa">
<td valign=bottom>&nbsp;<br>
<font color="#ffffff" face="helvetica, arial">&nbsp;<br><big><big><strong>TypeError</strong></big></big></font></td
><td align=right valign=bottom
><font color="#ffffff" face="helvetica, arial">Python 3.8.10: /usr/bin/python3<br>Mon Nov 27 12:57:38 2023</font></td></tr></table>
    
<p>A problem occurred in a Python script.  Here is the sequence of
function calls leading up to the error, in the order they occurred.</p>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///home/aaitouna/Webserv/public/Python/index.py">/home/aaitouna/Webserv/public/Python/index.py</a> in <strong>&lt;module&gt;</strong></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;&nbsp;7</small>&nbsp;form&nbsp;=&nbsp;cgi.FieldStorage()<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;&nbsp;8</small>&nbsp;#&nbsp;Get&nbsp;data&nbsp;from&nbsp;fields<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;&nbsp;&nbsp;9</small>&nbsp;nm=form.getvalue('name')<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;10</small>&nbsp;f=form.getvalue('faculty')<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;11</small>&nbsp;b=form.getvalue('batch')<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090">nm <em>undefined</em>, <strong>form</strong>&nbsp;= FieldStorage(None, None, ''), form.<strong>getvalue</strong>&nbsp;= &lt;bound method FieldStorage.getvalue of FieldStorage(None, None, '')&gt;</font></small></td></tr></table>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///usr/lib/python3.8/cgi.py">/usr/lib/python3.8/cgi.py</a> in <strong>getvalue</strong>(self=FieldStorage(None, None, ''), key='name', default=None)</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;531</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;def&nbsp;getvalue(self,&nbsp;key,&nbsp;default=None):<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;532</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"""Dictionary&nbsp;style&nbsp;get()&nbsp;method,&nbsp;including&nbsp;'value'&nbsp;lookup."""<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;533</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;key&nbsp;in&nbsp;self:<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;534</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;value&nbsp;=&nbsp;self[key]<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;535</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;isinstance(value,&nbsp;list):<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090"><strong>key</strong>&nbsp;= 'name', <strong>self</strong>&nbsp;= FieldStorage(None, None, '')</font></small></td></tr></table>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///usr/lib/python3.8/cgi.py">/usr/lib/python3.8/cgi.py</a> in <strong>__contains__</strong>(self=FieldStorage(None, None, ''), key='name')</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;571</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"""Dictionary&nbsp;style&nbsp;__contains__&nbsp;method."""<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;572</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;self.list&nbsp;is&nbsp;None:<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;573</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;TypeError("not&nbsp;indexable")<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;574</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;any(item.name&nbsp;==&nbsp;key&nbsp;for&nbsp;item&nbsp;in&nbsp;self.list)<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;575</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090"><em>builtin</em> <strong>TypeError</strong>&nbsp;= &lt;class 'TypeError'&gt;</font></small></td></tr></table><p><strong>TypeError</strong>: not indexable
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>args&nbsp;=
('not indexable',)
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>with_traceback&nbsp;=
&lt;built-in method with_traceback of TypeError object&gt;


<!-- The above is a description of an error in a Python program, formatted
     for a Web browser because the 'cgitb' module was enabled.  In case you
     are not reading this in a Web browser, here is the original traceback:

Traceback (most recent call last):
  File "public/Python/index.py", line 9, in &lt;module&gt;
    nm=form.getvalue('name')
  File "/usr/lib/python3.8/cgi.py", line 533, in getvalue
    if key in self:
  File "/usr/lib/python3.8/cgi.py", line 573, in __contains__
    raise TypeError("not indexable")
TypeError: not indexable

-->

